src/main/webapp/WEB-INF/pages/errorProductNotFound.jsp

```diff
@@ -0,0 +1,17 @@
+<%@ page import="com.es.phoneshop.model.product.ProductNotFoundException" %>
+<%@ page contentType="text/html;charset=UTF-8" %>
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+<%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
+<%@ taglib prefix="tags" tagdir="/WEB-INF/tags" %>
+
+<%! ProductNotFoundException exception;%>
+<% exception = (ProductNotFoundException) request.getAttribute("javax.servlet.error.exception");
+   request.setAttribute("prodId", exception.getProductId());
+%>
```

Comment:
Please try to avoid plain java code in .jsp files. Use different approach to show exception data